### PR TITLE
Do not apply text color if it is being overriden

### PR DESCRIPTION
**Pull request by webmandesign**
Fixing #24625

### CODE REVIEW

File: packages/block-library/src/button/style.scss [hunk 1]
```diff
@@ -49,7 +49,11 @@ $blocks-button__height: 56px;
 
 .is-style-outline .wp-block-button__link,
 .wp-block-button__link.is-style-outline {
-	color: $dark-gray-700;
 	background-color: transparent;
 	border: 2px solid;
 }
+
+.is-style-outline .wp-block-button__link:not(.has-text-color),
```
Comment (ZebulanStanphill) [inline packages/block-library/src/button/style.scss:56]: To reduce potential of style conflicts:
```suggestion
.is-style-outline > .wp-block-button__link:not(.has-text-color),
```

(See also: #24919.)

Comment (webmandesign) [reply packages/block-library/src/button/style.scss]: Thanks. I've updated the file and applied your fix where needed.

**Comment by jorgefilipecosta** [comment]
Hi @webmandesign,

I'm sorry I started an alternative PR because I did not notice you had PR. Apparently using the "Fixing" keyword does not link an issue to the PR. Fix, fixes, etc do the linking making it more easier to notice a PR has an issue, the full list of keywords one can use to link PR's to issues is available at https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords#about-issue-references.


This PR needed a rebase, so I rebased and I kept the color #32373c. This color is essentially a default and I think removing it and using inherit may be considered a breaking change. So given that we are very close to 5.6 release I would prefer not to have this risk.


Regarding the background color fix I proposed in https://github.com/WordPress/gutenberg/pull/26707, I think it is still needed because although the outline does not have a background by default the user is able to select one:
image

Without the fix in https://github.com/WordPress/gutenberg/pull/26707, the background selection may not work.

So I'm merging this PR which was the first one to fix the text color issue, and then merging https://github.com/WordPress/gutenberg/pull/26707 that fixes the background issue.


Thank you for reporting and fixing this issue 👍

**Comment by webmandesign** [comment]
Hi @jorgefilipecosta,

Sorry for the keyword, I'll know for the future.

Thanks for merging the PR. It's really unfortunate that Gutenberg forces the color here. As a theme author I need to override it in styles unnecessary, in my opinion. To me it makes perfect sense that outlined button inherit a text color of the container. Forcing a specific color (which does not comes from a theme) is very confusing to users.

Seeing this color I was also thinking about Gutenberg color audit. It seems there might be multiple instances of similar color forcing.

As for background color I understand your point. But from my perspective the outline styling has no specific meaning then. If a default button styling in the theme is the one as in your image, user can recreate such button even if the style is selected to be outline. Which is pretty confusing to me. Outline should remove any background. That's how I understand it. But I don't see this as big of an issue as the forced text color :) 

Thanks and regards,

Oliver

**Pull request merged**